hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AtttributeResolver.kt

```diff
@@ -51,37 +30,44 @@ private fun findAttributes(
   prefix: String = "",
   nested: Boolean = false,
 ): List<Pair<String, Attribute>> =
-  when (getPropertyType(prop)) {
-    PropertyType.SIMPLE -> {
-      getPropertyClass(prop).let { propClass ->
-        val propertyName = prefix + prop.name
-        val openSearchName = prefix + getOpenSearchName(prop, propClass)
-        val fuzzy = FUZZY_ATTRIBUTES.contains(propertyName)
-        listOf(propertyName to Attribute(propClass, openSearchName, nested, fuzzy))
-      }
+  when (prop.getPropertyType()) {
```

Comment:
This class is full of cryptic framework styled code, so has been littered with comments to try and give people half a chance if they ever need to change it